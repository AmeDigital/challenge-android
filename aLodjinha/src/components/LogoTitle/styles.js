import { StyleSheet } from 'react-native';
import { RFValue } from 'react-native-responsive-fontsize';

import material from '../../utils/MaterialDesign';

export default StyleSheet.create({
	image: {
		width: 30,
		height: 30,
		marginHorizontal: 15
	},
	textTile: {
		color: material.colorWhiteTwo,
		fontSize: RFValue(24),
		letterSpacing: 0.6,
		fontFamily: material.fontPacificoRegular
	}
});
