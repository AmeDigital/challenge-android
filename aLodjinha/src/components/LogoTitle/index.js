import React from 'react';
import { Image, Text } from 'react-native';

import logo from '../../images/logo_navbar.png';
import styles from './styles';
import Icon from 'react-native-vector-icons/MaterialIcons';

export default function LogoTitle({ navigation }) {
	return (
		<React.Fragment>
			<Icon name="menu" style={{ marginLeft: 15 }} onPress={navigation.openDrawer} size={24} color="#fff" />
			<Image source={logo} style={styles.image} />
			<Text style={styles.textTile}>a Lodjinha</Text>
		</React.Fragment>
	);
}
