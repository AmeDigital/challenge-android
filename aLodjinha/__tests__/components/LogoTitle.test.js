import React from 'react';

import renderer from 'react-test-renderer';

import LogoTitle from '../../src/components/LogoTitle';

describe('Component LogoTitle', () => {
	test('should render without erros', () => {
		const tree = renderer.create(<LogoTitle navigation={{ openDrawer: function() {} }} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
