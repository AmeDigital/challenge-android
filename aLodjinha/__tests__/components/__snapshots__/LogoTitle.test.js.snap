// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LogoTitle should render without erros 1`] = `
Array [
  <Text
    allowFontScaling={false}
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 24,
        },
        Object {
          "marginLeft": 15,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î—’
  </Text>,
  <Image
    source={
      Object {
        "testUri": "../../../src/images/logo_navbar.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginHorizontal": 15,
        "width": 30,
      }
    }
  />,
  <Text
    style={
      Object {
        "color": "#fcfcfc",
        "fontFamily": "Pacifico-Regular",
        "fontSize": 47,
        "letterSpacing": 0.6,
      }
    }
  >
    a Lodjinha
  </Text>,
]
`;
